src/features/diagnosticsProvider.ts

```diff
@@ -208,12 +206,13 @@ class DiagnosticsProvider extends AbstractSupport {
         // This check is just small perf optimization to reduce queries
         // for omnisharp with analyzers (which has event to notify about updates.)
         if (!this._analyzersEnabled) {
-            this._validateAllPipe.next();
+            this._validateAllPipe.next("onDocumentOpenOrChange");
         }
     }
 
     private _onProjectAnalysis(event: protocol.ProjectDiagnosticStatus) {
-        if (event.Status == DiagnosticStatus.Ready) {
+        if (event.Status == DiagnosticStatus.Ready &&
+            event.ProjectFilePath === "(100 %)") {
```

Comment:
We are currently getting Ready status updates from O# with "(0 %)" regularly. This was causing us to continually request updated diagnostics.